common/src/main/java/com/wynntils/models/lootrun/LootrunModel.java

```diff
@@ -107,6 +111,14 @@ public class LootrunModel extends Model {
     //                        ÀÀÀ§7Challenges Completed: §f7
 
     private static final Pattern LOOTRUN_FAILED_PATTERN = Pattern.compile("[À\\s]*§c§lLootrun Failed!");
+    private static final Pattern CHALLENGE_FAILED_PATTERN = Pattern.compile("[À\\s]*§c§lChallenge Failed!");
+
+    private static final Pattern MISSION_COMPLETED_PATTERN = Pattern.compile("[À\\s]*§b§lMission Completed");
+
+    // These two prefixes are used to identify when a mission is completed,
+    // as this information is presented in multiple ways
+    private static final String ADVANCED_MISSION_PREFIX = ".*[À\\\\s]*§b§l";
+    private static final String MISSION_PREFIX = ".*?§.";
```

Comment:
So can you explain a bit more about these patterns? I need lines for them, because these regexes are too broad. Why the ".*" in both cases? Is one of the patterns for completion, one for mission start? 

Also, you could create the regex in-line with something like `Pattern.compile("...(" + Arrays.stream(MissionType.values()).map(MissionType::getName).collect(Collectors.joining("|")) + ")");`

Reply:
To be honest, I'm not too familiar with regex, so most of these patterns are taken from other parts of the project with some assistance from AI development tools. 

When a mission is completed, it can be presented in one of two ways:

-   It is completed during a challenge, presenting the "Mission Complete" message before displaying the mission name.
-   It is completed as a part of a challenge's finishing phase, where it is presented in a format without the "Mission Complete" text and where the mission name is always the Bolded Aqua color. 

Each of these regex patterns covers its corresponding case.

Currently, this feature does not track when missions are started, as I did not feel it was necessary due to the current mission already being tracked on the scoreboard along with its completion requirement.

Reply:
Okay. Can you send the chat messages from some logs then? Only then, I can help with the regex.

Reply:
First case:
`[CHAT]                          ÀÀ&b&lMission Completed
18:18:40.301
wynntils
Render thread
[CHAT]                     ÀÀÀ&7You have finished 1 mission!
18:18:40.302
wynntils
Render thread
[CHAT]                                   ÀÀÀ&eStasis
18:18:40.302
wynntils
Render thread
[CHAT]                    À&7While choosing a beacon, your
18:18:40.302
wynntils
Render thread
[CHAT]                 À&7timer does not decrease (Max 5m)
`

Second case:
`[CHAT]            À&7Parsimonious&r                      ÀÀ&b&lMaterialism
17:43:09.708
wynntils
Render thread
[CHAT]                    À                   ÀÀÀ&7All Challenges now additionally
17:43:09.708
wynntils
Render thread
[CHAT]                    À                  ÀÀ&7have the Yellow Beacon Effect.
17:43:09.708
wynntils
Render thread
[CHAT]                    À                   &7Yellow Beacons cannot appear
17:43:09.708
wynntils
Render thread
[CHAT]                    À                                 À&7anymore`

Reply:
[patterns.txt](https://github.com/user-attachments/files/15864484/patterns.txt)